torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -4546,6 +4546,147 @@ Tensor cumprod_jvp(Tensor self_t, Tensor self_p, Tensor result, int dim) {
   }
 }
 
+Tensor batch_norm_jvp(
+    const Tensor& input_p, const Tensor& input_t,
+    const Tensor& weight_p, const Tensor& weight_t,
+    const Tensor& bias_p, const Tensor& bias_t,
+    const c10::optional<Tensor>& running_mean,
+    const c10::optional<Tensor>& running_var,
+    const Tensor& saved_mean, const Tensor& saved_invstd,
+    bool train,
+    double eps) {
+  auto dims = std::vector<int64_t>{};
+  auto view_size = input_t.sizes().vec();
+  int64_t numel = 1;
+  for (const auto dim : c10::irange(view_size.size())) {
+    if (dim != 1) {
```

Comment:
batchnorm doesn't support no-batch dim?

Reply:
Looking at https://github.com/pytorch/pytorch/issues/60585
- yes, batch norm semantically is incompatible with no-batch dim
- for group norm it is BC-breaking, so a potential plan seems to be to create a new  `nn.Module` entirely that would handle the no batch dim case
- layer norm should be fine as long as we handle the case where normalized shape has the same dimension as input. I do see a test case in OpInfo handling this so we should be ok

cc @jbschlosser